internal/controller/ngrok/kubernetesoperator_controller.go

```diff
@@ -116,17 +117,23 @@ func (r *KubernetesOperatorReconciler) Reconcile(ctx context.Context, req ctrl.R
 }
 
 func (r *KubernetesOperatorReconciler) create(ctx context.Context, ko *ngrokv1alpha1.KubernetesOperator) error {
+	var k8sOp *ngrok.KubernetesOperator
+	var ngrokErr ngrokgo.Error
+
 	k8sOp, err := r.findExisting(ctx, ko)
 	if err != nil {
 		return err
 	}
 
+	// defer status update
+	defer func() {
+		// best effort
+		_ = r.updateStatus(ctx, ko, k8sOp, ngrokErr)
+	}()
+
```

Comment:
I've thought about this a bit more and can we hold off on using this new defer pattern for updating the statuses?

Reply:
Yes, and I want to do that with both status PRs merged so I can have 1 PR to overhaul the logic.